src/comm.cpp

```diff
@@ -230,13 +246,16 @@ void nb_allreduce_aluminum(El::Matrix<T, El::Device::GPU>& m,
                            typename BackendT::allreduce_algo_type algo
                            = BackendT::allreduce_algo_type::automatic) {
   const auto local_size = m.Height() * m.Width();
+  const auto& syncinfo = El::SyncInfoFromMatrix(m);
+  auto& request = GetRequest(req, tag);
   ::Al::NonblockingAllreduce<BackendT>(
     m.Buffer(),
     local_size,
     mpi_op_to_al_op(op),
-    c.template GetComm<BackendT>(El::SyncInfoFromMatrix(m)),
-    GetRequest(req, tag),
+    c.template GetComm<BackendT>(syncinfo),
+    request,
     algo);
+  UpdateRequest(request, syncinfo);
```

Comment:
Would it be better to put this before the allreduce, just in case Aluminum does something funky when the `req_type`'s stream is wrong?

Reply:
No; the request gets created as part of the call to Aluminum. (For Allreduce, e.g., [this line](https://github.com/LLNL/Aluminum/blob/3f4170fce069eef773f25142665afc16fb5d3ee5/src/nccl_impl.hpp#L206) calls the common `setup_completion_event()` function, which [creates the request](https://github.com/LLNL/Aluminum/blob/3f4170fce069eef773f25142665afc16fb5d3ee5/src/nccl_impl.hpp#L354). The `UpdateRequest()` line just resets the stream to the right thing.